gravitee-apim-console-webui/src/portal/customization/theme/portal-theme.component.ts

```diff
@@ -14,32 +14,214 @@
 import { MatCardModule } from '@angular/material/card';
-import { ReactiveFormsModule, Validators, FormControl, FormGroup } from '@angular/forms';
+import { ReactiveFormsModule, FormControl, FormGroup } from '@angular/forms';
 import { MatFormFieldModule } from '@angular/material/form-field';
 import { MatInputModule } from '@angular/material/input';
-import { GioSaveBarModule } from '@gravitee/ui-particles-angular';
+import { GioAvatarModule, GioFormFilePickerModule, GioMonacoEditorModule, GioSaveBarModule, NewFile } from '@gravitee/ui-particles-angular';
+import { MatExpansionModule } from '@angular/material/expansion';
+import { MatIconModule } from '@angular/material/icon';
+import { MatAnchor, MatButton } from '@angular/material/button';
+import { MatTooltipModule } from '@angular/material/tooltip';
+import { catchError, map, takeUntil, tap } from 'rxjs/operators';
+import { EMPTY, Subject } from 'rxjs';
+import { isEqual } from 'lodash';
 
+import { GioFormColorInputModule } from '../../../shared/components/gio-form-color-input/gio-form-color-input.module';
 import { PortalHeaderComponent } from '../../components/header/portal-header.component';
+import { UiPortalThemeService } from '../../../services-ngx/ui-theme.service';
+import { ThemePortalNext, UpdateThemePortalNext } from '../../../entities/management-api-v2';
+import { SnackBarService } from '../../../services-ngx/snack-bar.service';
+
+export interface ThemeVM {
+  logo?: string[];
+  favicon?: string[];
+  font?: string;
+  primaryColor?: string;
+  secondaryColor?: string;
+  tertiaryColor?: string;
+  errorColor?: string;
+  pageBackgroundColor?: string;
+  cardBackgroundColor?: string;
+  customCSS?: string;
+}
 
 @Component({
   selector: 'portal-theme',
   templateUrl: './portal-theme.component.html',
   styleUrls: ['./portal-theme.component.scss'],
-  imports: [CommonModule, GioSaveBarModule, MatCardModule, MatFormFieldModule, MatInputModule, ReactiveFormsModule, PortalHeaderComponent],
+  imports: [
+    CommonModule,
+    GioFormFilePickerModule,
+    GioSaveBarModule,
+    MatCardModule,
+    MatFormFieldModule,
+    MatIconModule,
+    MatInputModule,
+    ReactiveFormsModule,
+    MatExpansionModule,
+    GioFormColorInputModule,
+    GioAvatarModule,
+    GioFormFilePickerModule,
+    MatAnchor,
+    MatButton,
+    MatTooltipModule,
+    GioMonacoEditorModule,
+    PortalHeaderComponent,
+  ],
   standalone: true,
 })
-export class PortalThemeComponent {
-  public developerPortalThemeForm = new FormGroup({
-    name: new FormControl(null, Validators.required),
-  });
+export class PortalThemeComponent implements OnInit, OnDestroy {
+  constructor(
+    private readonly uiPortalThemeService: UiPortalThemeService,
+    private readonly snackBarService: SnackBarService,
+  ) {}
+
+  private initialTheme: ThemePortalNext;
+  private defaultValues: ThemeVM;
+  private initialValues: ThemeVM;
+
+  public portalThemeForm;
```

Comment:
Non-blocking comment: Best to use a typed FormGroup for a form. It saves quite a bit of headache in the long run and is easier to debug.